### PR TITLE
OSDOCS-3147: Adding EUS to EUS upgrade doc

### PR DESCRIPTION
**Pull request by sagidlow**
Applies to 4.8+ (Will need to create a separate PR for 4.9). This PR will CP to 4.8 and 4.10. 
JIRA: https://issues.redhat.com/browse/OSDOCS-3147
SME and QE ack required. 
Preview Links:

- [Preparing to perform an EUS to EUS upgrade](https://deploy-preview-40619--osdocs.netlify.app/openshift-enterprise/latest/updating/preparing-eus-eus-upgrade.html)
- [EUS to EUS upgrade](https://deploy-preview-40619--osdocs.netlify.app/openshift-enterprise/latest/updating/preparing-eus-eus-upgrade.html#updating-eus-to-eus-upgrade_eus-to-eus-upgrade)

### CODE REVIEW

**Comment by sagidlow** [comment]
@sdodson @wking @LalatenduMohanty Could you all PTAL? Also, could you recommend someone from QE to take a look at this as well.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+----
+$ oc get clusterversion
+----
++
+.Example output
++
+[source,terminal]
+----
+NAME  	  VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
+version   4.10.1	  True    	  False     	  6m29s   Cluster version is 4.10.1
```
Comment (sagidlow) [inline modules/updating-eus-to-eus-upgrade.adoc:113]: @sdodson - Could you verify this example output? There wasn't any example output in the Google Doc, but I based it off the 4.9.11 above.

File: updating/preparing-eus-eus-upgrade.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[id="preparing-eus-eus-upgrade"]
+= Performing to perform an EUS to EUS upgrade
+include::modules/common-attributes.adoc[]
+:context: eus-to-eus-upgrade
+
+toc::[]
+
+Due to fundamental Kubernetes design, all {product-title} upgrades between minor versions must be serialized.
+You must upgrade from 4.8 to 4.9 and then to 4.10. You cannot upgrade from 4.8 to 4.10 directly.
+However, beginning the upgrade from 4.8 to 4.9 to 4.10, administrators who wish to upgrade between two Extended Update Support (EUS) versions can do so incurring only a single reboot of non-master hosts.
```
Comment (sdodson) [inline updating/preparing-eus-eus-upgrade.adoc:10]: However, beginning with...

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 1 above)
Comment (sdodson) [reply modules/updating-eus-to-eus-upgrade.adoc:113]: That's fine.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+The following procedure pauses all non-master MachineConfigPools and performs upgrades from 4.8 to 4.9 to 4.10, then unpauses the previously paused MachineConfigPools.
+Following this procedure reduces the total upgrade duration and the number of times workloads are restarted.
+
+.Prerequisites
+
+* Review the release notes for {product-title} 4.9 and 4.10
+* Ensure that you are familiar with version-specific prerequisites, such as Administrator Acknowledgement that is required prior to upgrading from {product-title} 4.8 to 4.9.
+* Verify that your cluster is not running {product-title} 4.8.14 version or earlier.
+If your cluster is running {product-title} 4.8.14 or earlier, you must upgrade to a later 4.8.z version prior to upgrading to 4.9.
+The upgrade to 4.8.14 or later fulfills the minimum version requirements that must be performed without pausing MachineConfigPools.
```
Comment (sdodson) [inline modules/updating-eus-to-eus-upgrade.adoc:17]:
```suggestion
The upgrade to 4.8.14 or later which is necessary to fulfill the minimum version requirements must be performed without pausing MachineConfigPools.
```
Or something like that, just a bit hard to follow as is.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 4]
```diff
@@ -0,0 +1,144 @@
+// Module included in the following assemblies:
+//
+// * updating/preparing-eus-eus-upgrade.adoc
+
+[id="updating-eus-to-eus-upgrade_{context}"]
+= EUS to EUS upgrade
+
+The following procedure pauses all non-master MachineConfigPools and performs upgrades from 4.8 to 4.9 to 4.10, then unpauses the previously paused MachineConfigPools.
+Following this procedure reduces the total upgrade duration and the number of times workloads are restarted.
```
Comment (LalatenduMohanty) [inline modules/updating-eus-to-eus-upgrade.adoc:9]: s/workloads/worker nodes/

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+.Prerequisites
+
+* Review the release notes for {product-title} 4.9 and 4.10
+* Ensure that you are familiar with version-specific prerequisites, such as Administrator Acknowledgement that is required prior to upgrading from {product-title} 4.8 to 4.9.
```
Comment (LalatenduMohanty) [inline modules/updating-eus-to-eus-upgrade.adoc:14]: Should we link the admin doc URL here?

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 5 above)
Comment (sagidlow) [reply modules/updating-eus-to-eus-upgrade.adoc:14]: Add link: https://docs.openshift.com/container-platform/4.9/updating/updating-cluster-prepare.html#update-preparing-migrate_updating-cluster-prepare

**Comment by sagidlow** [comment]
@jiajliu could you please review?

File: updating/preparing-eus-eus-upgrade.adoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+You must upgrade from 4.8 to 4.9 and then to 4.10. You cannot upgrade from 4.8 to 4.10 directly.
+However, beginning with the upgrade from 4.8 to 4.9 to 4.10, administrators who wish to upgrade between two Extended Update Support (EUS) versions can do so incurring only a single reboot of non-master hosts.
+
+There are a number of caveats to consider when attempting an EUS to EUS upgrade.
+
+* EUS to EUS upgrades is only offered after upgrades between all versions involved have been made available in `stable` channels.
+* If you encounter issues during or after upgrading to the odd-numbered minor version but before upgrading to the next even-numbered version, then remediation of those issues may require that non-master hosts complete the upgrade to the odd-numbered version before moving forward.
+* You can complete the upgrade process during multiple maintenance windows by pausing at intermediate steps. However, plan to complete the entire upgrade within 60 days. This is critical to ensure that normal cluster automation processes are completed including those associated with certificate rotation.
```
Comment (jiajliu) [inline updating/preparing-eus-eus-upgrade.adoc:16]: Generally sounds reasonable to suggest the whole update process's duration limit. But i'm not sure certificates rotation cycle. @xingxingxia @wangke19 Could you help confirm if 60 days meet the certificate rotate's requirement and any risks here? Thanks!

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+. Verify that all MachineConfigPools display a status of `UPDATED` and no MachineConfigPools display a status of `UPDATING`.
+To view the status of all MachineConfigPools, run the following command:
++
+[source,terminal]
+----
+$ oc get mcp
+----
++
+.Example output
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:30]: This line seems indented. Refer to https://github.com/openshift/openshift-docs/blob/840ecfcd8214b9109d0abfd415e488c0fdcd53c5/modules/updating-eus-to-eus-upgrade.adoc

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
++
+[NOTE]
+====
+You cannot pause the master pool.
+====
++
+[source,terminal]
+----
+$ oc patch mcp/worker --type merge --patch '{"spec":{"paused":true}}'
+$ oc get mcp -o
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:51]: The command is split to two lines wrongly.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+master      false
+worker      true
+----
+
+. To change to the `eus-4.10` channel and upgrade to 4.9, run the following commands:
++
+[source,terminal]
+----
+$ oc adm upgrade channel eus-4.10
+$ oc adm upgrade --to-latest
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:69]: When it's a disconnected cluster, we can not do upgrade with --to-latest directly. Maybe we could give a link here for users to redirect to upgrade in disconnected cluster section.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+. To pause the MachineConfigPools you wish to skip reboots on, run the following commands:
++
+[NOTE]
+====
+You cannot pause the master pool.
+====
++
+[source,terminal]
+----
+$ oc patch mcp/worker --type merge --patch '{"spec":{"paused":true}}'
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:50]: How about add the output of the patch command?
```
# ./oc patch mcp/worker --type merge --patch '{"spec":{"paused":true}}'
machineconfigpool.machineconfiguration.openshift.io/worker patched
```

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+.Example output
++
+Output is trimmed for clarity:
++
+[source,terminal]
+----
+NAME 	   CONFIG                                            UPDATED     UPDATING
+master   rendered-master-ecbb9582781c1091e1c9f19d50cf836c    True  	 False
+worker   rendered-worker-00a3f0c68ae94e747193156b491553d5    True 	 False
+----
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:144]: We could give a new mc config here(in contrast the old one before update in step 1).  
```
master   rendered-master-52da4d2760807cb2b96a3402179a9a4c
worker   rendered-worker-4756f60eccae96fb9dcb4c392c69d497
```

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+NAME  	  VERSION  AVAILABLE  PROGRESSING   SINCE   STATUS
+version   4.10.1   True       False         6m29s   Cluster version is 4.10.1
+----
+
+. To unpause all previously paused MachineConfigPools, run the following command:
++
+[source,terminal]
+----
+$ oc patch mcp/worker --type merge --patch '{"spec":{"paused":false}}'
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:120]: The same here.
```
# ./oc patch mcp/worker --type merge --patch '{"spec":{"paused":false}}'
machineconfigpool.machineconfiguration.openshift.io/worker patched
```

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+[id="updating-eus-to-eus-upgrade_{context}"]
+= EUS to EUS upgrade
+
+The following procedure pauses all non-master MachineConfigPools and performs upgrades from 4.8 to 4.9 to 4.10, then unpauses the previously paused MachineConfigPools.
+Following this procedure reduces the total upgrade duration and the number of times worker nodes are restarted.
+
+.Prerequisites
+
+* Review the release notes for {product-title} 4.9 and 4.10
```
Comment (sdodson) [inline modules/updating-eus-to-eus-upgrade.adoc:13]:
```suggestion
* Review the release notes for {product-title} 4.9 and 4.10
* Review the release notes and product lifecycles for any layered products and OLM Operators, some may require updates either before or during an EUS-to-EUS upgrade
```

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+* Ensure that you are familiar with version-specific prerequisites, such as link:https://docs.openshift.com/container-platform/4.9/updating/updating-cluster-prepare.html#update-preparing-migrate_updating-cluster-prepare[administrator acknowledgement] that is required prior to upgrading from {product-title} 4.8 to 4.9.
+* Verify that your cluster is not running {product-title} 4.8.14 version or earlier.
+If your cluster is running {product-title} 4.8.14 or earlier, you must upgrade to a later 4.8.z version prior to upgrading to 4.9.
+The upgrade to 4.8.14 or later which is necessary to fulfill the minimum version requirements that must be performed without pausing MachineConfigPools.
+* Verify that MachineConfigPools is unpaused.
+
+.Procedure
+
```
Comment (sdodson) [inline modules/updating-eus-to-eus-upgrade.adoc:21]:
```suggestion
. Upgrade any OLM Operators to versions which are compatible with both versions you will be upgrading to.
```

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 7 above)
Comment (sagidlow) [reply modules/updating-eus-to-eus-upgrade.adoc:30]: @jiajliu Yes, it is. It is meant to group things together.

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 8 above)
Comment (sagidlow) [reply modules/updating-eus-to-eus-upgrade.adoc:51]: @jiajliu not sure what you mean? @sdodson is this not how it Is supposed to be?

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 7 above)
Comment (jiajliu) [reply modules/updating-eus-to-eus-upgrade.adoc:33]: ack

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 8 above)
Comment (jiajliu) [reply modules/updating-eus-to-eus-upgrade.adoc:54]: Yes, i mean the command is split to line 54 and 64, which should be together in one line.

File: updating/preparing-eus-eus-upgrade.adoc (hunk 6 above)
Comment (jiajliu) [reply updating/preparing-eus-eus-upgrade.adoc:16]: ack, then no concern from my side. thanks for the explanation!

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 9 above)
Comment (jiajliu) [reply modules/updating-eus-to-eus-upgrade.adoc:75]: make sense. ack.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+[source,terminal]
+----
+$ oc patch mcp/worker --type merge --patch '{"spec":{"paused":false}}'
+----
++
+.Example output
++
+[source,terminal]
+----
+# ./oc patch mcp/worker --type merge --patch '{"spec":{"paused":false}}'
```
Comment (jiajliu) [inline modules/updating-eus-to-eus-upgrade.adoc:135]: @sagidlow Looks like we could remove line 135 if we keep line 128. They are duplicated.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+Following this procedure reduces the total upgrade duration and the number of times worker nodes are restarted.
+
+.Prerequisites
+
+* Review the release notes for {product-title} 4.9 and 4.10
+* Review the release notes and product lifecycles for any layered products and OLM Operators, some may require updates either before or during an EUS to EUS upgrade.
+* Ensure that you are familiar with version-specific prerequisites, such as link:https://docs.openshift.com/container-platform/4.9/updating/updating-cluster-prepare.html#update-preparing-migrate_updating-cluster-prepare[administrator acknowledgement] that is required prior to upgrading from {product-title} 4.8 to 4.9.
+* Verify that your cluster is not running {product-title} 4.8.14 version or earlier.
+If your cluster is running {product-title} 4.8.14 or earlier, you must upgrade to a later 4.8.z version prior to upgrading to 4.9.
+The upgrade to 4.8.14 or later which is necessary to fulfill the minimum version requirements that must be performed without pausing MachineConfigPools.
```
Comment (sdodson) [inline modules/updating-eus-to-eus-upgrade.adoc:18]:
```suggestion
The upgrade to 4.8.14 or later which is necessary to fulfill the minimum version requirements must be performed without pausing MachineConfigPools.
```

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+----
+$ oc adm upgrade channel eus-4.10
+$ oc adm upgrade --to-latest
+----
++
+.Example output
++
+[source,terminal]
+----
+Updating to latest version 4.9.11
```
Comment (sdodson) [inline modules/updating-eus-to-eus-upgrade.adoc:68]:
```suggestion
Updating to latest version 4.9.18
```

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+----
+$ oc get clusterversion
+----
++
+.Example output
++
+[source,terminal]
+----
+NAME  	  VERSION  AVAILABLE  PROGRESSING   SINCE   STATUS
+version   4.9.11   True       False         6m29s   Cluster version is 4.9.11
```
Comment (sdodson) [inline modules/updating-eus-to-eus-upgrade.adoc:83]:
```suggestion
version   4.9.18   True       False         6m29s   Cluster version is 4.9.18
```

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 17 above)
Comment (sdodson) [reply modules/updating-eus-to-eus-upgrade.adoc:68]: We've recently become aware of an issue which required us to bump the minimum version required to upgrade from 4.9 to 4.10. This probably doesn't matter but I don't want someone seeing 4.9.11 here and questioning why they can't upgrade to 4.10 from 4.9.11.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+[id="updating-eus-to-eus-upgrade_{context}"]
+= EUS to EUS upgrade
+
+The following procedure pauses all non-master MachineConfigPools and performs upgrades from 4.8 to 4.9 to 4.10, then unpauses the previously paused MachineConfigPools.
+Following this procedure reduces the total upgrade duration and the number of times worker nodes are restarted.
+
+.Prerequisites
+
+* Review the release notes for {product-title} 4.9 and 4.10
+* Review the release notes and product lifecycles for any layered products and OLM Operators, some may require updates either before or during an EUS to EUS upgrade.
```
Comment (nalhadef) [inline modules/updating-eus-to-eus-upgrade.adoc:14]: Should be two sentences (period after "OLM Operators." Or a semi-colon, if you like those things. :-)

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 16 above)
Comment (nalhadef) [inline modules/updating-eus-to-eus-upgrade.adoc:18]: * Line 17: Is "z" necessary in "4.8.z?" It's already specified that a version later than 4.8.14 is needed.
* Line 18: Is a negative overlap created when saying to upgrade to 4.8.14 or later when Line 17 says that 4.8.14 won't work?

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+----
+$ oc patch mcp/worker --type merge --patch '{"spec":{"paused":false}}'
+----
++
+[NOTE]
+====
+If pools are not unpaused, the cluster is not permitted to upgrade to any future minors and maintenance tasks such as certificate rotation are inhibited. This puts the cluster at risk for future degradation.
+====
+
+. To verify that your previously paused pools have updated and you cluster completed the upgrade to 4.10, run the following command:
```
Comment (nalhadef) [inline modules/updating-eus-to-eus-upgrade.adoc:122]: Should "you cluster" be "your cluster?"

File: updating/preparing-eus-eus-upgrade.adoc [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+toc::[]
+
+Due to fundamental Kubernetes design, all {product-title} upgrades between minor versions must be serialized.
+You must upgrade from 4.8 to 4.9 and then to 4.10. You cannot upgrade from 4.8 to 4.10 directly.
+However, beginning with the upgrade from 4.8 to 4.9 to 4.10, administrators who wish to upgrade between two Extended Update Support (EUS) versions can do so incurring only a single reboot of non-master hosts.
+
+There are a number of caveats to consider when attempting an EUS to EUS upgrade.
+
+* EUS to EUS upgrades is only offered after upgrades between all versions involved have been made available in `stable` channels.
+* If you encounter issues during or after upgrading to the odd-numbered minor version but before upgrading to the next even-numbered version, then remediation of those issues may require that non-master hosts complete the upgrade to the odd-numbered version before moving forward.
```
Comment (nalhadef) [inline updating/preparing-eus-eus-upgrade.adoc:15]: Wow! This is like how the odd numbered Star Trek movies are weaker than the even numbered ones. :-)

File: updating/preparing-eus-eus-upgrade.adoc [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+
+toc::[]
+
+Due to fundamental Kubernetes design, all {product-title} upgrades between minor versions must be serialized.
+You must upgrade from 4.8 to 4.9 and then to 4.10. You cannot upgrade from 4.8 to 4.10 directly.
+However, beginning with the upgrade from 4.8 to 4.9 to 4.10, administrators who wish to upgrade between two Extended Update Support (EUS) versions can do so incurring only a single reboot of non-master hosts.
+
+There are a number of caveats to consider when attempting an EUS to EUS upgrade.
+
+* EUS to EUS upgrades is only offered after upgrades between all versions involved have been made available in `stable` channels.
```
Comment (nalhadef) [inline updating/preparing-eus-eus-upgrade.adoc:14]: Should "is" be "are?"

File: updating/preparing-eus-eus-upgrade.adoc [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+* If you encounter issues during or after upgrading to the odd-numbered minor version but before upgrading to the next even-numbered version, then remediation of those issues may require that non-master hosts complete the upgrade to the odd-numbered version before moving forward.
+* You can complete the upgrade process during multiple maintenance windows by pausing at intermediate steps. However, plan to complete the entire upgrade within 60 days. This is critical to ensure that normal cluster automation processes are completed including those associated with certificate rotation.
+* You must be running at least 4.8.14 before starting the EUS-to-EUS upgrade procedure. If you do not meet this minimum requirement, upgrade to a later 4.8.z before attempting the EUS-to-EUS upgrade.
```
Comment (nalhadef) [inline updating/preparing-eus-eus-upgrade.adoc:17]: Another conflict with what's above, where the version had to be later than 4.8.14. (unless I'm confusing different things.)

File: modules/updating-eus-to-eus-upgrade.adoc (hunk 16 above)
Comment (sagidlow) [reply modules/updating-eus-to-eus-upgrade.adoc]: I adjusted the wording across the board to help with this issue.

File: modules/updating-eus-to-eus-upgrade.adoc [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+----
++
+.Example output
++
+[source,terminal]
+----
+Updating to latest version 4.9.18
+----
+
+. To ensure the 4.9 upgrades is completed successfully retrieve the cluster version, run the following command:
```
Comment (nalhadef) [inline modules/updating-eus-to-eus-upgrade.adoc:71]: Ooops. Missed this the first runthrough. "Is" should be "are."
